lib/bin/lwm2m_carrier/doc/app_integration.rst

```diff
@@ -69,105 +87,114 @@ See :file:`nrf\\lib\\bin\\lwm2m_carrier\\include\\lwm2m_carrier.h` for all the e
+  * If this event persists, contact the carrier.
+
+* :c:macro:`LWM2M_CARRIER_DEFERRED_SERVER_CONNECT`:
+
+  * This event indicates that the DTLS handshake with the server has failed.
+  * This typically happens if the bootstrap sequence has failed on the carrier side.
+
+* :c:macro:`LWM2M_CARRIER_DEFERRED_SERVER_REGISTRATION`:
+
+  * The server registration has not completed, and the server does not recognize the connecting device.
+  * If this event persists, contact the carrier.
+
+* :c:macro:`LWM2M_CARRIER_EVENT_FOTA_START`:
+
+  * This event indicates that the modem update has started.
+  * The application must immediately terminate any open TLS sessions.
+  * See :ref:`req_appln_limitations`.
+
+* :c:macro:`LWM2M_CARRIER_EVENT_REBOOT`:
+
+  * If the application is not ready to reboot, it must return non-zero and then reboot at the earliest convenient time.
+
+* :c:macro:`LWM2M_CARRIER_EVENT_ERROR`:
+
+  * This event indicates an error.
+  * The event data struct :c:type:`lwm2m_carrier_event_error_t` contains the information about the error (:c:member:`code` and :c:member:`value`).
+  * Following are the valid error codes:
+
+    * :c:macro:`LWM2M_CARRIER_ERROR_CONNECT_FAIL` - This error is generated from the :c:func:`lte_lc_init_and_connect` function. It indicates possible problems with the SIM card, or insufficient network coverage. See :c:member:`value` field of the event.
+
+    * :c:macro:`LWM2M_CARRIER_ERROR_DISCONNECT_FAIL` - This error is generated from the :c:func:`lte_lc_offline` function. See :c:member:`value` field of the event.
+
+    * :c:macro:`LWM2M_CARRIER_ERROR_BOOTSTRAP` - This error is generated from the :c:func:`modem_key_mgmt_write` function, if the :c:member:`value` field is negative. If the :c:member:`value` field is 0, it indicates that the bootstrap sequence has failed. If this error persists, contact your carrier.
+
+    * :c:macro:`LWM2M_CARRIER_ERROR_FOTA_PKG` - This error indicates that the update package has been rejected. The integrity check has failed because of a wrong package sent from the server, or a wrong package received by client. The :c:member:`value` field will have an error of type :c:type:`nrf_dfu_err_t` from the file :file:`nrfxlib\\nrf_modem\\include\\nrf_socket.h`.
+
+    * :c:macro:`LWM2M_CARRIER_ERROR_FOTA_PROTO` - This error indicates a protocol error. There might be unexpected HTTP header contents. The server might not support partial content requests.
+
+    * :c:macro:`LWM2M_CARRIER_ERROR_FOTA_CONN` - This error indicates a connection problem. Either the server host name could not be resolved, or the remote server refused the connection.
+
+    * :c:macro:`LWM2M_CARRIER_ERROR_FOTA_CONN_LOST` - This error indicates a loss of connection, or an unexpected closure of connection by the server.
+
+    * :c:macro:`LWM2M_CARRIER_ERROR_FOTA_FAIL` - This error indicates a failure in applying a valid update. If this error persists, create a ticket in `DevZone`_ with the modem trace.
+
```

Comment:
`LWM2M_CARRIER_ERROR_SERVICE_UNAVAILABLE` and `LWM2M_CARRIER_ERROR_CONFIGURATION` seem to be missing.